File: backend/Chef.DAL/Entities/Recipe.cs
using System;
using System.Collections.Generic;
using Chef.DAL.Entities.Common;

namespace Chef.DAL.Entities
{
    public class Recipe : AuditEntity
    {
        public Recipe()
        {
            Components = new List<RecipeComponent>();
            Instructions = new List<Instruction>();
            Reviews = new List<Review>();
        }

        public string Name { get; set; }
        public string Description { get; set; }
        public TimeSpan? CookingTime { get; set; }
        public TimeSpan? PreparationTime { get; set; }
        public int DefaultServingsCount { get; set; }
        public int CategoryId { get; set; }
        public RecipeCategory Category { get; set; }
        public int LikeCount { get; set; }
        public int DislikeCount { get; set; }
        public ICollection<RecipeComponent> Components { get; set; }
        public ICollection<Instruction> Instructions { get; set; }
        public ICollection<Review> Reviews { get; set; }
    }
}

File: frontend/src/app/shared/models/recipe/recipe.ts
import { Time } from "@angular/common";
import { Component } from "../component/component";
import { Instruction } from "../instruction/instruction";
import { Review } from "../review/review";

export interface Recipe {
    id: number;
    name: string;
    description: string;
    cookingTime: Time;
    preparationTime: Time;
    recipeUrl: string;
    likeCount: number;
    defaultServingsCount: number;
    dislikeCount: number;
    summaryTime: Time;
    ingredientsCount: number;
    reviews: Review[];
    instructions: Instruction[];
    components: Component[];
}